### ISSUE TITLE
atom feed 504 gateway timeout

### ISSUE THREAD

**Issue by judell** [hypothesis/h#2329]
Seems sensitive to the amount of data returned. A less-restrictive query:

```
curl -L "http://hypothes.is/stream.atom?user=judell"
<html>
 <head>
  <title>504 Gateway Timeout</title>
 </head>
 <body>
  <h1>504 Gateway Timeout</h1>
  The gateway has timed out.<br/><br/>
HTTPSConnectionPool(host='hypothes.is', port=443): Read timed out. (read timeout
=0.2)
```

A more-restrictive query:

```
curl -L "http://hypothes.is/stream.atom?user=judell&tags=reading" 
<? xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
````

(We saw this in stage, speculated it was a stage artifact, evidently not.)

**Comment by tilgovi** [comment]
Probably deployment then. We may need to bump the proxy read timeout in nginx.

**Comment by tilgovi** [comment]
Another thing would be to investigate whether we can generate a streaming response by templating each item separately, rather than iterating in the template, and returning a `Response` object with a generator as the body iterator.

**Comment by seanh** [comment]
The app itself has a (configurable) timeout when waiting for the api to respond - I suspect that rather than the nginx timeout. Double that setting will fix it I think

**Comment by tilgovi** [comment]
You're almost certainly right. The default proxy_upstream_timeout in nginx is 60s. The default API client timeout in h is 0.2s.

**Comment by tilgovi** [comment]
I would probably do more than double it. The cost of having the connection open is low, especially with greenlets for each request. If the client disconnects, I think the greenlet exits anyway, so these aren't going to hang around. Something like 2s might be more appropriate.

**Comment by seanh** [comment]
> I would probably do more than double it. The cost of having the connection open is low, especially with greenlets for each request. If the client disconnects, I think the greenlet exits anyway, so these aren't going to hang around. Something like 2s might be more appropriate.

Seems fine to me. I don't see why it should be set low. Rather wait a couple secs for a page to load, than get a 504. (Of course we should make the app fast as well.)

**Comment by tilgovi** [comment]
Closed in 653f19e3b6a14db542432752ba3cc43e7c0f8ffe
